lib/set-script.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+const parseJSON = require('json-parse-even-better-errors')
+const rpj = require('read-package-json-fast')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function (args) {
+  if (process.env.npm_lifecycle_event === 'postinstall')
+    log.error('Scripts can\'t set from the postinstall script')
+
```

Comment:
i don't believe this will actually abort the operation - it will just log an error.
```suggestion
  if (process.env.npm_lifecycle_event === 'postinstall') {
    log.error('Scripts can\'t set from the postinstall script')
    throw 'Scripts can\'t set from the postinstall script' // or maybe an error here?
  }
```